crates/bitwarden/src/mobile/vault/client_ciphers.rs

```diff
@@ -17,7 +17,18 @@ impl<'a> ClientCiphers<'a> {
 
         // TODO: Once this flag is removed, the key generation logic should
         // be moved directly into the KeyEncryptable implementation
-        if cipher_view.key.is_none() && self.client.get_flags().enable_cipher_key_encryption {
+
+        // If any attachments have no keys, they will need to be reuploaded before we can
+        // generate them a cipher key, so in that case we skip them even if the flag is set
+        let all_attachments_have_keys = cipher_view
+            .attachments
+            .iter()
+            .flatten()
+            .all(|a| a.key.is_some());
+        if cipher_view.key.is_none()
+            && self.client.get_flags().enable_cipher_key_encryption
+            && all_attachments_have_keys
+        {
```

Comment:
I believe the other clients do re-encrypt ciphers with attachments. They just leave attachments keys as null.

Reply:
Ah okay, I changed it in 817c9fffe978b6cd024766fb7613940827455c33.

Now when encrypting a cipher we always leave null attachment keys as null (so it continues using the org/user key). When the keys are not null we reencrypt as usual.

In the move to org case we still need to error out if the keys are null.

Does that sound about right?